tests/io/test_ase.py

```diff
@@ -297,6 +297,17 @@ def test_back_forth(filename):
     for key, val in atoms.todict().items():
         assert str(atoms_back.todict()[key]) == str(val)
 
+    atoms = ase.io.read(f"{TEST_FILES_DIR}/{filename}")
+    mask = [True] * len(atoms)
+    mask[-1] = False
+    atoms.set_constraint(ase.constraints.FixAtoms(mask=mask))
+    structure = AseAtomsAdaptor.get_structure(atoms)
+    atoms_back = AseAtomsAdaptor.get_atoms(structure)
+    structure_back = AseAtomsAdaptor.get_structure(atoms_back)
+    assert structure_back == structure
+    for key, val in atoms.todict().items():
+        assert str(atoms_back.todict()[key]) == str(val)
```

Comment:
Could you define this as a new test function since it is not dependent on the prior block of code?

Reply:
Done